jetty-util/src/main/java/org/eclipse/jetty/util/thread/ReservedThreadExecutor.java

```diff
@@ -296,37 +302,12 @@ private Runnable reservedWait()
                 if (!isRunning())
                     return STOP;
 
-                boolean idle = false;
-                try (Locker.Lock lock = _locker.lock())
+                try
                 {
-                    if (_task == null)
-                    {
-                        try
-                        {
-                            if (_idleTime == 0)
-                                _wakeup.await();
-                            else
-                                idle = !_wakeup.await(_idleTime, _idleTimeUnit);
-                        }
-                        catch (InterruptedException e)
-                        {
-                            LOG.ignore(e);
-                        }
-                    }
-                    else
-                    {
-                        Runnable task = _task;
-                        _task = null;
-
-                        if (LOG.isDebugEnabled())
-                            LOG.debug("{} task={}", this, task);
-
+                    Runnable task = _idleTime == 0 ? _task.take() : _task.poll(_idleTime, _idleTimeUnit);
+                    if (task != null)
```

Comment:
There was a `LOG.debug()` statement when we had a non-null `task` that I would like to keep.